code/config/test/Test/Loot/Config.hs

```diff
@@ -10,9 +10,10 @@
 
 module Test.Loot.Config where
```

Comment:
I think it would be worth to add some roundtrip tests to check that a serialized configuration can be deserialized.

Possibly these tests could be taken from that [previous PR](https://github.com/serokell/lootbox/pull/53/files#diff-18574cf2f7db45bc4758fd8f958241c83db00a7bd1833155c2468571c7082df1R273-R314) without changes.

Reply:
I'll do it

Reply:
Hm... I've added this tests and one test fail, but not at every launch!

```
jsonRoundtripOptionPartial: FAIL (0.10s)
        ✗ jsonRoundtripOptionPartial failed at test/Test/Loot/Config.hs:285:3
          after 59 tests and 12 shrinks.
        
              ┏━━ test/Test/Loot/Config.hs ━━━
          282 ┃ hprop_jsonRoundtripOptionPartial :: Property
          283 ┃ hprop_jsonRoundtripOptionPartial = property $ do
          284 ┃   config <- forAll cfgOptionPartial
              ┃   │ {str =: "\55296", int =: 0, sub =< {int2 <unset>, bool <unset>, sub2 =< {str2 <unset>, mem <unset>}}, kek <unset>, tre =+ {treType <unset>, str3 <unset>, brc1 =- {int3 <unset>}, brc2 =- {str4 <unset>, sub3 =< {int4 <unset>}}}}
          285 ┃   (eitherDecode . encode) config === Right config
              ┃   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              ┃   │ Failed
              ┃   │ ━━ lhs ━━
              ┃   │ Right {str =: "\65533", int =: 0, sub =< {int2 <unset>, bool <unset>, sub2 =< {str2 <unset>, mem <unset>}}, kek <unset>, tre =+ {treType <unset>, str3 <unset>, brc1 =- {int3 <unset>}, brc2 =- {str4 <unset>, sub3 =< {int4 <unset>}}}}
              ┃   │ ━━ rhs ━━
              ┃   │ Right {str =: "\55296", int =: 0, sub =< {int2 <unset>, bool <unset>, sub2 =< {str2 <unset>, mem <unset>}}, kek <unset>, tre =+ {treType <unset>, str3 <unset>, brc1 =- {int3 <unset>}, brc2 =- {str4 <unset>, sub3 =< {int4 <unset>}}}}
```

Reply:
Interesting, it looks like some generated strings are causing issues.
From the error message it appears that in the roundtrip of the `"\55296"` `String` something goes wrong.

It looks like that happens in the encoding, because checking in ghci
```
import Data.Aeson
import qualified Data.ByteString.Lazy.Char8 as LBC8

LBC8.putStrLn $ encode "\55296"
```
seems to produce gibberish for me.

I'm not sure if this is intended behavior or there's an issue in `aeson`, this should be checked (see [this issue](https://github.com/haskell/aeson/issues/304)).

Either way we should avoid the problem (modifying the `string` generator, I guess).

Reply:
Hm... maybe this is the reason, why `String` was changed to `Text` at #53

Reply:
I've changed gen to text, and in cfg building I call `unpack`

Reply:
That seems to work, even manually testing for it :ok_hand: 

However please add a comment in `cfgOptionPartial` to explain why we use `Gen.text`.